### PR TITLE
Added problem_026

### PR DESCRIPTION
**Pull request by mansiag**
**Fixes issue:**
N/A

**Changes:**
Added ProjectEuler problem 26 in c++

### CODE REVIEW

File: code/online_challenges/src/project_euler/problem_026/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Project Euler Problem #026: Reciprocal cycles  
+
+([Problem Link](https://projecteuler.net/problem=26))
+
+
+
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/README.md]: While they make no different to the output, I would remove all of these spaces.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#include <bits/stdc++.h>
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: #1915

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+using namespace std;
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Avoid for naming conflicts.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+int main ()
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Remove spaces before parentheses.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+{
+
+   int remainder, value, position, sequenceLength = 0;
+
+   for (int i = 1000; i > 0; i--)
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Pre decrement.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+   {
+
+       int cycleCheckArray[i];
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Use `vector`s, not raw arrays.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+       for (int j = 0; j < i; j++)
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Fix indentation; Use pre increment.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+	       cycleCheckArray[j] = 0;
+
+       remainder = 1, value = 1, position = 0;
+
+       while (1)
+	   {
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: indentation

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+   {
+
+       int cycleCheckArray[i];
+
+       for (int j = 0; j < i; j++)
+	       cycleCheckArray[j] = 0;
+
+       remainder = 1, value = 1, position = 0;
+
+       while (1)
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: true is better than 1

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+	    }
+
+	    cycleCheckArray[remainder] = position;
+	    position++;
+
+   }
+
+       if (sequenceLength == i - 1)
+       {
+	       cout << i;
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Add a newline (`\n`). Don't use `endl`.

**Comment by mansiag** [comment]
@arnavb Can you please review the changes?. Thanks.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+#include <vector>
+int main()
+{
+
+   int remainder, value, position, sequenceLength = 0;
+
+   for (int i = 1000; i > 0; --i)
+   {
+
+       std::vector <int> cycleCheckArray;
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Remove space before less-than sign.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+       }
+
+       if (sequenceLength == i - 1)
+       {
+	       std::cout << i <<"\n";
+	       return 0;
+       }
+   }
+}
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Use spaces for indent.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+       std::vector <int> cycleCheckArray;
+
+       for (int j = 0; j < i; ++j)
+	       cycleCheckArray.push_back(0);
+
+       remainder = 1, value = 1, position = 0;
+
+       while (true)
+	   {
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Excess spaces.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+	       remainder = value % i;
+
+	       if (cycleCheckArray[remainder])
+	       {
+	           sequenceLength = position - cycleCheckArray[remainder];
+	           break;
+	       }
+
+	       cycleCheckArray[remainder] = position;
+	       position++;
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Use `++position`.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 15]
```diff
@@ -0,0 +1,41 @@
+#include <iostream>
+#include <vector>
+int main()
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: I would add a newline before main for separation. Otherwise, good job! 👍

**Comment by mansiag** [comment]
@iattempt Can you please review the changes?

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp (hunk 15 above)
Comment (mansiag) [reply code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Thanks

**Comment by iattempt** [comment]
@mansiag Please use Ctrl+F to distinguish the space and the tab and use 4 spaces as indentation.

**Comment by mansiag** [comment]
@iattempt Is it alright now?

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 16]
```diff
@@ -0,0 +1,42 @@
+#include <iostream>
+#include <vector>
+
+int main()
+{
+
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Remove this line.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+        std::vector<int> cycleCheckArray;
+
+        for (int j = 0; j < i; ++j)
+            cycleCheckArray.push_back(0);
+
+        remainder = 1, value = 1, position = 0;
+        while (true)
+        {
+
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Same here.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+            if (cycleCheckArray[remainder])
+            {
+                sequenceLength = position - cycleCheckArray[remainder];
+                break;
+            }
+
+            cycleCheckArray[remainder] = position;
+            ++position;
+
```
Comment (iattempt) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: And this.

File: code/online_challenges/src/project_euler/problem_026/problem_026.cpp [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        if (sequenceLength == i - 1)
+        {
+            std::cout << i <<"\n";
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_026/problem_026.cpp]: Add a space before the ".

**Comment by mansiag** [comment]
@arnavb Can you please review the changes?

**Comment by mansiag** [comment]
@adichat Can you please review this pull request?

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**